vllm/sequence.py

```diff
@@ -688,7 +688,7 @@ def maybe_set_first_token_time(self, time: float) -> None:
         #   in TPOT, rather than recalculating TTFT (since from the )
         #   POV of the user, there is simply a long generation delay.
         if (self.metrics.first_token_time is None
-                and next(iter(self.seqs)).get_output_len() == 1):
+                and self.seqs[0].get_output_len() == 1):
```

Comment:
This is one of the mlperf micro optimizations

Reply:
I'll do some profiling, was looking at the following upstream PRs and their comments
https://github.com/vllm-project/vllm/pull/7051
https://github.com/vllm-project/vllm/pull/6520